### ISSUE TITLE
"Some issues need your attention" error on video send

### ISSUE THREAD

**Issue by TheHiddenPaw** [signalapp/Signal-Android#8573]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When sending media (video) file, the file doesn't send and Signal delivers a fairly vague error message: "Some issues need your attention" Suspect from log it may be a file size issue, but it'd be useful if the error message reflected that...

### Steps to reproduce
- attach large media file (video)
- hit send
- on receipt of error message, attempt resend 
- Repeat ad infinitum

**Actual result:** Video will not send 

**Expected result:** Video sent, or useful error message

### Device info
**Device:** Sony E5823 (E5823)
**Android version:** Android : 7.1.1 (3761073091, 32.4.A.1.54)
**Signal version:** 4.32.8 

### Link to debug log
https://debuglogs.org/345154534e7ba1f2c1386b8908980f693abfd1ad94e8490dd85b372afbdf06da

**Comment by greyson-signal** [comment]
Yeah, so it looks like the file is, in fact, too large:

```java
01-30 18:25:21.978 19014 19204 W Job     : [1314aa11-3d7b-4b7c-87db-39d364869990] AttachmentUploadJob :: Failing due to an exception. (Time since submission: 194 ms, Run attempt: 0, isStopped: false)
01-30 18:25:21.978 19014 19204 W Job     : org.thoughtcrime.securesms.transport.UndeliverableMessageException: Size constraints could not be met!
01-30 18:25:21.978 19014 19204 W Job     : 	at org.thoughtcrime.securesms.jobs.AttachmentUploadJob.scaleAndStripExif(AttachmentUploadJob.java:142)
01-30 18:25:21.978 19014 19204 W Job     : 	at org.thoughtcrime.securesms.jobs.AttachmentUploadJob.onRun(AttachmentUploadJob.java:86)
01-30 18:25:21.978 19014 19204 W Job     : 	at org.thoughtcrime.securesms.jobmanager.Job.doWorkInternal(Job.java:111)
01-30 18:25:21.978 19014 19204 W Job     : 	at org.thoughtcrime.securesms.jobmanager.Job.doWork(Job.java:64)
01-30 18:25:21.978 19014 19204 W Job     : 	at androidx.work.Worker$1.run(Worker.java:84)
01-30 18:25:21.978 19014 19204 W Job     : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
01-30 18:25:21.978 19014 19204 W Job     : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
01-30 18:25:21.978 19014 19204 W Job     : 	at java.lang.Thread.run(Thread.java:761)
```

I agree this isn't a good experience. Let me figure out what the best recourse is here. 

Ideally, you shouldn't even be able to attempt to send an attachment that's too large, so that's one thing. Can you tell me how you attached the file? Thanks!

**Comment by TheHiddenPaw** [comment]
Certainly! 

- From either the bare message thread or after having typed a message, tap the "+" button to attach a file. (screenshot 1)
![1](https://user-images.githubusercontent.com/3433099/52011018-d3653900-24ce-11e9-88c7-5886704abb4e.png)
- Select 'Gallery' (screenshot 2)
![2](https://user-images.githubusercontent.com/3433099/52011023-d5c79300-24ce-11e9-8568-5ecd2cfd84cd.png)
- Select the file. Even if it's too big, it still opens in the 'play media' view, where you can write a message and send (screenshot 3)
![3](https://user-images.githubusercontent.com/3433099/52011026-d6602980-24ce-11e9-9b8e-6ecf660bc62f.png)
- Send, which fails, and when you click to find out why, it gives the unhelpful error (screenshot 4)
![4](https://user-images.githubusercontent.com/3433099/52011030-d7915680-24ce-11e9-9df3-595db1ef9529.png)

**Comment by jcvernaleo** [comment]
I thought the MMS size limit was pretty low (less than 3M or so but carrier dependent).  I believe that many MMS clients compress the video since almost no video fits below that limit (just checked and a 9s video on my google pixel is 51M).  I know the Messages app in AOSP compresses images at least (`Messaging/src/com/android/messaging/datamodel/data/MessagePartData.java` and `src/com/android/messaging/util/ImageUtils.java`) but that is usually missing things from the actual released versions.

**Comment by survivor303** [comment]
sure in mms but in signal messages, surely people dosnt want use unsecure and expensive sms/mms services :)

**Comment by greyson-signal** [comment]
This is fixed in the 4.34 release branch. Thanks!
